### PR TITLE
[FLAVA] Make projections part of the core model

### PR DESCRIPTION
**Pull request by ankitade**
Move projections from the contrastive loss to the core model
This will allow users to use the model (instead of the pretraining model) for doing zero shot
Also moved to using the translated the checkpoint.

Test plan
1. pytest
2.  python -m flava.train config=flava/configs/pretraining/debug.yaml
3. python -m flava.finetune config=flava/configs/finetuning/qnli.yaml
 
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #195
* #132
* #131
* __->__ #106
* #105

Differential Revision: [D37481127](https://our.internmc.facebook.com/intern/diff/D37481127)

### CODE REVIEW

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Review by ebsmothers — APPROVED** [review]

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37481127).
